mantidproject/mantid
# Project Recovery - Catch top level exceptions

**Cannot be merged before #22887**

**Description of work.**
Adds a top level try / catch around the recovery attempt. This will open a dialog box informing the user a full recovery could not be completed and that Mantid will continue running correctly.

**To test:**
- Open Mantid
- Create a sample workspace
- Ensure a checkpoint is saved
- Crash Mantid
- On the most recent checkpoint, open a .py file
- Remove the timestamp (everything after the # char)
- Restart and attempt recovery

Partly deals with #22901.

Does this update require release notes?

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
